cmd/livepeer/livepeer.go

```diff
@@ -355,57 +424,41 @@ func setupTranscoder(n *core.LivepeerNode, acct accounts.Account) (ethereum.Subs
 	rm := core.NewRewardManager(time.Second*5, n.Eth)
 	go rm.Start(context.Background())
 
-	//Subscribe to when a job is assigned to us
-	logsCh := make(chan ethtypes.Log)
-	sub, err := n.Eth.SubscribeToJobEvent(context.Background(), logsCh)
-	if err != nil {
-		glog.Errorf("Error subscribing to job event: %v", err)
-	}
-	// go func() error {
-	// select {
-	// case l := <-logsCh:
-	// tx, _, err := n.Eth.Backend().TransactionByHash(context.Background(), l.TxHash)
-	// if err != nil {
-	// 	glog.Errorf("Error getting transaction data: %v", err)
-	// }
-	// strmId, tData, err := eth.ParseJobTxData(tx.Data())
-	// if err != nil {
-	// 	glog.Errorf("Error parsing job tx data: %v", err)
-	// }
-
-	// jid, _, _, _, err := eth.GetInfoFromJobEvent(l, n.Eth)
-	// if err != nil {
-	// 	glog.Errorf("Error getting info from job event: %v", err)
-	// }
-
-	// //Create Transcode Config
-	// tProfiles := txDataToVideoProfile(tData)
-	// config := net.TranscodeConfig{StrmID: strmId, Profiles: tProfiles, JobID: jid, PerformOnchainClaim: true}
-	// glog.Infof("Transcoder got job %v - strmID: %v, tData: %v, config: %v", tx.Hash(), strmId, tData, config)
-
-	// //Do The Transcoding
-	// cm := core.NewClaimManager(strmId, jid, tProfiles, n.Eth)
-	// strmIDs, err := n.TranscodeAndBroadcast(config, cm)
-	// if err != nil {
-	// 	glog.Errorf("Transcode Error: %v", err)
-	// }
-
-	// //Notify Broadcaster
-	// sid := core.StreamID(strmId)
-	// vids := make(map[core.StreamID]types.VideoProfile)
-	// for i, vp := range tProfiles {
-	// 	vids[strmIDs[i]] = vp
-	// }
-	// if err = n.NotifyBroadcaster(sid.GetNodeID(), sid, vids); err != nil {
-	// 	glog.Errorf("Notify Broadcaster Error: %v", err)
-	// }
-
-	// 	return nil
-
-	// }
-	// }()
-
-	return sub, nil
+	go func() error {
+		select {
+		case l := <-logsCh:
+			_, _, jid := eth.ParseNewJobLog(l)
+
+			job, err := n.Eth.GetJob(jid)
+			if err != nil {
+				glog.Errorf("Error getting job info: %v", err)
+			}
+
+			//Create Transcode Config
+			tProfiles := txDataToVideoProfile(job.TranscodingOptions)
+			config := net.TranscodeConfig{StrmID: job.StreamId, Profiles: tProfiles, JobID: jid, PerformOnchainClaim: true}
+			glog.Infof("Transcoder got job %v - strmID: %v, tData: %v, config: %v", jid, job.StreamId, job.TranscodingOptions, config)
+
+			//Do The Transcoding
+			cm := core.NewClaimManager(job.StreamId, jid, tProfiles, n.Eth)
+			strmIDs, err := n.TranscodeAndBroadcast(config, cm)
+			if err != nil {
+				glog.Errorf("Transcode Error: %v", err)
+			}
+
+			//Notify Broadcaster
+			sid := core.StreamID(job.StreamId)
+			vids := make(map[core.StreamID]types.VideoProfile)
+			for i, vp := range tProfiles {
+				vids[strmIDs[i]] = vp
+			}
+			if err = n.NotifyBroadcaster(sid.GetNodeID(), sid, vids); err != nil {
+				glog.Errorf("Notify Broadcaster Error: %v", err)
+			}
+
+			return nil
+		}
+	}()
 }
 
 func txDataToVideoProfile(txData string) []types.VideoProfile {
```

Comment:
We should add error checking here - it's possible the job's video profiles aren't defined on this node.